lib/vagrant-service-manager/binary_handlers/binary_handler.rb

```diff
@@ -0,0 +1,118 @@
+require 'net/http'
+
+module VagrantPlugins
+  module ServiceManager
+    class URLValidationError < Vagrant::Errors::VagrantError
+      error_key(:url_validation_error)
+    end
+
+    class BinaryHandler
+      BINARY_ARCHIVE_FORMATS = ['.tgz', '.tar.gz', '.gz', '.zip'].freeze
+      BINARY_NAME = {
+        docker: 'docker', openshift: 'oc'
+      }.freeze
+      VERSION_CMD = {
+        docker: "docker version --format '{{.Server.Version}}'",
+        openshift: "oc version | grep 'oc' | grep -oE '[0-9.]+'"
+      }.freeze
+      BINARY_REGEX = {
+        windows: { docker: %r{\/docker.exe$}, openshift: %r{oc.exe$} },
+        unix: { docker: %r{\/docker$}, openshift: %r{oc$} }
+      }.freeze
+      ARCHIVE_MAP = {
+        '.tgz' => 'Tar', '.tar.gz' => 'Tar', '.gz' => 'Tar', '.zip' => 'Zip'
+      }.freeze
+      LABEL = 'servicemanager.commands.install_cli.message'.freeze
+
+      attr_accessor :path, :version, :type, :url,
+                    :binary_exists, :skip_download, :archive_file_path
+
+      def initialize(machine, env, options)
+        @machine = machine
```

Comment:
interesting, the first two variables you align and then you stop? why? 
Never mind, don't push anything for now. I have several changes locally now.